src/material/grid-list/testing/grid-list-harness.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate} from '@angular/cdk/testing';
+import {ɵTileCoordinator as TileCoordinator} from '@angular/material/grid-list';
+import {GridListHarnessFilters, GridTileHarnessFilters} from './grid-list-harness-filters';
+import {MatGridTileHarness} from './grid-tile-harness';
+
+/** Harness for interacting with a standard `MatGridList` in tests. */
+export class MatGridListHarness extends ComponentHarness {
+  /** The selector for the host element of a `MatGridList` instance. */
+  static hostSelector = '.mat-grid-list';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a `MatGridListHarness`
+   * that meets certain criteria.
+   * @param options Options for filtering which dialog instances are considered a match.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: GridListHarnessFilters = {}): HarnessPredicate<MatGridListHarness> {
+    return new HarnessPredicate(MatGridListHarness, options);
+  }
+
+  /**
+   * Tile coordinator that is used by the "MatGridList" for computing
+   * positions of tiles. We leverage the coordinator to provide an API
+   * for retrieving tiles based on visual tile positions.
+   */
+  private _tileCoordinator = new TileCoordinator();
+
+  /** Gets all tiles of the grid-list. */
+  async getTiles(filters: GridTileHarnessFilters = {}): Promise<MatGridTileHarness[]> {
+    return await this.locatorForAll(MatGridTileHarness.with(filters))();
+  }
+
+  /** Gets the amount of columns of the grid-list. */
+  async getColumns(): Promise<number> {
+    return Number(await (await this.host()).getAttribute('cols'));
+  }
+
+  /**
+   * Gets a tile of the grid-list that is located at the given location.
+   * @param row Zero-based row index.
+   * @param column Zero-based column index.
+   */
+  async getTileAtPosition({row, column}: {row: number, column: number}):
```

Comment:
Initially I just had separate parameters for row and column index, but in tests it's way more obvious if we use a keyed object literal here. I don't feel strongly about.

Also: I looked into having a method like `getPosition` in the tile harness (so that it can be also filtered), but it looks like this is not possible since the tile harness needs a reference to the grid-list harness. Also "position" is a broad term in the context of a tile, since the tile can span over multiple "positions". So I think it makes sense to just leave it to the grid-list harness.

Reply:
sgtm